### PR TITLE
IGNITE-11944: OpencensusExporter should export Histogram metrics.

### PR DESCRIPTION
**Pull request by NSAmelchev**

### CODE REVIEW

File: modules/opencensus/src/main/java/org/apache/ignite/spi/metric/opencensus/OpenCensusMetricExporterSpi.java [hunk 1]
```diff
@@ -118,6 +120,26 @@
      */
     private Map<String, Measure> measures = new HashMap<>();
 
+    /** Cached histogram metrics intervals names. */
+    private Map<T2<String, long[]>, String[]> histogramNames = new HashMap<>();
```
Comment (nizhikov) [inline modules/opencensus/src/main/java/org/apache/ignite/spi/metric/opencensus/OpenCensusMetricExporterSpi.java]: Seems, a map is not cleared on `MetricRegistry` remove.

File: modules/opencensus/src/main/java/org/apache/ignite/spi/metric/opencensus/OpenCensusMetricExporterSpi.java [hunk 2]
```diff
@@ -183,6 +205,36 @@ else if (metric instanceof DoubleMetric) {
...
                     }
+                    else if (metric instanceof HistogramMetric) {
+                        /*
+                            Histogram exports as a separate long metric for each interval.
+                            Example of metric names if bounds are 10,100:
+                                histogram_0_10 (less than 10)
+                                histogram_10_100 (between 10 and 100)
+                                histogram_100_inf (more than 100)
+                         */
+                        long[] value = ((HistogramMetric)metric).value();
```
Comment (nizhikov) [inline modules/opencensus/src/main/java/org/apache/ignite/spi/metric/opencensus/OpenCensusMetricExporterSpi.java]: A variable should named `values`.

File: modules/opencensus/src/main/java/org/apache/ignite/spi/metric/opencensus/OpenCensusMetricExporterSpi.java [hunk 3]
```diff
@@ -118,6 +120,26 @@
...
+    private Map<T2<String, long[]>, String[]> histogramNames = new HashMap<>();
+
+    /** Create histogram interval names. */
+    private static final Function<T2<String, long[]>, String[]> CREATE_INTERVAL_NAMES = k -> {
+        String histogramName = k.get1();
+        long[] bounds = k.get2();
+
+        String[] names = new String[bounds.length + 1];
+
+        for (int i = 0; i < bounds.length + 1; i++) {
```
Comment (nizhikov) [inline modules/opencensus/src/main/java/org/apache/ignite/spi/metric/opencensus/OpenCensusMetricExporterSpi.java]: `i < bounds.length + 1` should be replaced with `i <= bounds.length`

Comment (NSAmelchev) [reply modules/opencensus/src/main/java/org/apache/ignite/spi/metric/opencensus/OpenCensusMetricExporterSpi.java]: Done.

File: modules/opencensus/src/main/java/org/apache/ignite/spi/metric/opencensus/OpenCensusMetricExporterSpi.java (hunk 2 above)
Comment (NSAmelchev) [reply modules/opencensus/src/main/java/org/apache/ignite/spi/metric/opencensus/OpenCensusMetricExporterSpi.java]: Done. Thank you.

File: modules/opencensus/src/main/java/org/apache/ignite/spi/metric/opencensus/OpenCensusMetricExporterSpi.java (hunk 1 above)
Comment (NSAmelchev) [reply modules/opencensus/src/main/java/org/apache/ignite/spi/metric/opencensus/OpenCensusMetricExporterSpi.java]: Done. Thank you.

**Pull request merged**